src/components/GroupAll/groupFilter/LanguageSelectBox.tsx

```diff
@@ -0,0 +1,325 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+import { IcArrowBottomGray, IcArrowTopGray, IcFilter } from '../../../assets';
+import CommonHashTag from '../../../common/CommonHashTag';
+import { LanguageSectionProps } from '../../../types/GroupCreate/GroupCreateType';
+import PageLayout from '../../PageLayout/PageLayout';
+import RangeSliderFilter from './RangeSlidrFilter';
+
+const ALL_TAG = 'ALL';
+const firstRowTags = ['Python', 'Java', 'JavaScript', 'C++', 'C', 'C#'];
+const secondRowTags = ['Kotlin', 'Swift', 'Ruby', 'Scala', 'Go'];
+
+const LanguageSelectBox = ({
+  selectedTags,
+  setSelectedTags,
+}: LanguageSectionProps) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  // 태그와 슬라이더의 상태를 별도로 관리
+  const [tempMinValue, setTempMinValue] = useState<number>(1);
+  const [tempMaxValue, setTempMaxValue] = useState<number>(50);
+  const [isSliderValueVisible, setIsSliderValueVisible] = useState(false); // 슬라이더 값 표시 여부
+
+  const toggleDropdown = () => {
+    setIsDropdownOpen((prev) => !prev);
+    if (!isSliderValueVisible) {
+      setIsSliderValueVisible(true); // 드롭다운이 처음 열릴 때 슬라이더 값을 표시
+    }
+  };
+
+  const addTag = (tag: string) => {
+    if (selectedTags.length >= 3 && tag !== ALL_TAG) {
+      return;
+    }
+
+    if (tag === ALL_TAG) {
+      selectAllTags();
+    } else {
+      if (!selectedTags.includes(tag)) {
+        const newTags = selectedTags.includes(ALL_TAG)
+          ? [...selectedTags.filter((t) => t !== ALL_TAG), tag]
+          : [...selectedTags, tag];
+
+        setSelectedTags(newTags);
+      }
+    }
+  };
+
+  const removeTag = (tag: string) => {
+    if (tag === ALL_TAG) {
+      setSelectedTags([]);
+    } else {
+      const newTags = selectedTags.filter((t) => t !== tag);
+      setSelectedTags(newTags);
+    }
+  };
+
+  const selectAllTags = () => {
+    const allTags = [...firstRowTags, ...secondRowTags];
+    setSelectedTags(allTags);
+  };
+
+  const onChangeMin = (value: number) => {
+    setTempMinValue(value);
+  };
+
+  const onChangeMax = (value: number) => {
+    setTempMaxValue(value);
+  };
```

Comment:
P2: 위에처럼 상태를 정의했다면 이렇게 줄일 수 있어요
```suggestion
  const onChangeTempValue = (value: number, isMinValue: boolean) => {
    setTempValue({
      ...tempValue,
      [isMinValue ? 'min' : 'max']: value,
    });
  };
```

Reply:
만약 상태 정의를 지금 형식대로 유지한다고 해도 아래처럼 함수를 바꿔주는게 더 깔끔할 것 같습니다!
```suggestion
  const onChangeTempValue = (value: number, isMinValue: boolean) => {
    isMinValue ? setTempMinValue(value) : setTempMaxValue(value);
  };
```

Reply:
넵! 수정하겠습니다